### ISSUE TITLE
replace/enhance isX patterns with SWRL rules, e.g. for phosphorylates

### ISSUE THREAD

**Issue by cmungall** [oborel/obo-relations#259]
Example:
http://purl.obolibrary.org/obo/RO_0002447

has axiom:

`(capable of o is kinase activity o has direct input) subPropertyOf phosphorylates`

this infers entity-entity precise relationships given a GO-CAM pattern

this pattern is familiar to owl ultra-geeks but is probably bamboozling to others. It doesn't help that ontobee doesn't show self-reflexivity axiom for http://purl.obolibrary.org/obo/RO_0002481. Another issue is that this is not in EL.

It would be cleaner to write this as a SWRL rule. We can have an optional transform to the self-reflexive pattern.

thoughts @balhoff ?

**Comment by cmungall** [comment]
```
?a1 enabled_by ?gp1
?gp1 a GO:'kinase activity'
?a1 directly-regulates ?a2
?a2 enabled-by ?gp2
->
?gp1 phosphorylates ?gp2
```

similar pattern for ubiquitination etc

**Comment by balhoff** [comment]
I agree the SWRL rule is less mysterious and would actually be supported in some GO reasoning situations (Arachne or Whelk). However it wouldn't contribute to classification; maybe this doesn't matter since the current Self axiom isn't either. `ObjectHasSelf` is [supposed to be in EL](https://www.w3.org/TR/2012/REC-owl2-profiles-20121211/#Class_Expressions), but as you say ELK doesn't support it (I checked 0.5 also).

**Comment by cmungall** [comment]
Good point. Still useful for a lot of ABox use cases.

**Comment by cthoyt** [comment]
I noticed that the only two post-translational modifications available in OBO-RO are phosphorylates and ubiquitinates. Is it non-trivial to add more of these relationships if they're being defined this way? 

I'm interested in this because we capture several more PTMs in BEL (amination, nitration, nitrosylation, glycation, etc.) that live under the GO term [cellular protein modification process (GO:0006464)](https://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0006464). I'm not (yet) an ultra-geek, but it seems like defining more of these using this SWRL rule might not be so difficult.

Should I make a new issue to continue this discussion?

**Comment by cmungall** [comment]
> Is it non-trivial to add more of these relationships if they're being defined this way?

Trivial but a bit tedious. I think we should try and devote a couple of hours to getting our templating framework set up so this becomes a matter of filling in terms in a CSV. @balhoff would dosdp-tools/spec require any work to include SWRL?

Yes, go ahead and make a new ticket, thanks!

**Comment by dosumis** [comment]
> ?a1 enabled_by ?gp1
> ?gp1 a GO:'kinase activity’
> ?a1 directly-regulates ?a2
> ?a2 enabled-by ?gp2
> ->
> ?gp1 phosphorylates ?gp2
> On 28 Sep 2018, at 01:34, Chris Mungall <notifications@github.com> wrote:
> 
> Is it non-trivial to add more of these relationships if they're being defined this way?
> 
> Trivial but a bit tedious. I think we should try and devote a couple of hours to getting our templating framework set up so this becomes a matter of filling in terms in a CSV. @balhoff <https://github.com/balhoff> would dosdp-tools/spec require any work to include SWRL?
> 
> 

It should be reasonably straightforward to extend the DOSDP spec to support adding rules.  Needs support for OP vars (deliberately omitted to now - but I should probably add these anyway).  Do you anticipate adding very large numbers of these?  Will they be self-contained or part of templates specifying a broader pattern?

**Comment by nlharris** [comment]
Is this still needed (and if so, who could work on it)?

**Comment by cthoyt** [comment]
This is related to #522, which will get some discussion at the RO meeting next week

**Comment by cthoyt** [comment]
I'm still interested in this
